forte/data/ontology/core.py

```diff
@@ -429,7 +435,20 @@ def _resolve_pointer(self, ptr: BasePointer) -> Entry:
         if isinstance(ptr, Pointer):
             return self.pack.get_entry(ptr.tid)
         elif isinstance(ptr, MpPointer):
-            return self.pack.packs[ptr.pack_index].get_entry(ptr.tid)
+            # bugfix/new feature 559: in new version pack_index will be using pack_id internally
+            pack_array_index = ptr.pack_index  # old version
+            pack_version = ""
+            try:
+                pack_version = self.pack.pack_version
+            except AttributeError:
+                pack_version = DEFAULT_PACK_VERSION  # set to default if lacking version attribute
+
+            if Version(pack_version) >= Version(BACKWARD_COMPATIBLE_VER):
```

Comment:
Similar to the other comment, `BACKWARD_COMPATIBLE_VER` is to board, it didn't indicate what we are supporting backward. We should rename it and provide a comment for it.

Reply:
This one is Version/Base_pack related. So coordinated with 500/547 related changes, and in the current checked in code as requested the name changed to "PACK_ID_COMPATIBLE_VERSION" ,  the same change (name) will also be in 500/547 related pull requests which resolves version specific issue. Future version related requests will be re-directed to these version specific changes in order to make each bug less dependent on others/more focused.